doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,313 @@
+- imagename: quay.io/operator-framework/scorecard-kuttltests:v0.0.1
+  description: redhat - kuttl tests
+  labels:
+    suite: kuttl
+```
+
+Users could override the default test images shipped with scorecard
+by adding the following into the scorecard DSL configuration:
+```
+default-images: 
+- imagetype: olm
+  image: quay.io/operator-framework/scorecard-olmtests:v0.0.1
+- imagetype: basic
+  image: quay.io/operator-framework/scorecard-basictests:v0.0.1
+- imagetype: kuttl
+  image: quay.io/operator-framework/scorecard-kuttltests:v0.0.1
+tests:
+```
+
+### Custom Test Execution
+
+Scorecard tests would be executed from the SDK scorecard CLI as is the case today.  Scorecard would execute the tests, capture the results, and provide feedback to the end-user by means of the current scorecard API (e.g. ScorecardTestResult).  Command line flags would allow the end-user a means to specify which tests are executed and upon which test environment to run the tests.  For example:
+```
+operator-sdk scorecard -o text --selector=suite=isvcustom --bundle='some/bundle/image:v0.0.1'
+```
+
+Running the above command would have scorecard perform the following:
+ 
+ * fetch the scorecard DSL yaml configuration from the bundle image
+ * select the tests to run based on the scorecard DSL configuration and the selector
+ * iterate through each test, 
+   * construct a test Pod 
+   * create the test Pod
+   * when the test pod completes, fetch the test image Pod log output for aggregation of test results
+ * display the test results 
+
+### Custom Test Environments
+
+Scorecard would allow you to run tests on a local Kube cluster as is the case today.  Provisioning the test cluster is external to the scope of the scorecard itself.  Note, that with this proposal, scorecard would not be responsible for installing OLM or depend on OLM being installed in order to run. 
+
+
+#### Migrating Scorecard Internal Tests to Custom Test Images
+
+With the above custom test capability, there is benefit to Red Hat in migrating the existing scorecard internal tests to be based on the custom test format. 
+
+Today there are 2 internal scorecard tests that make use of the scorecard proxy, with these tests migrating to the custom test format, the scorecard proxy is likely not necessary which would also reduce the overall scorecard implementation complexity.
+
+### Upgrade / Downgrade Strategy
+
+The changes to scorecard are large and would require a migration off
+of the current scorecard.
+
+## Drawbacks
+
+
+## Ansible and Helm Operator Support
+
+### Ansible Operator Complex Tests
+
+For custom tests, it might be possible to write the custom test image using ansible's callback plugin framework to produce container image log output in the scorecard required format (e.g. v1alpha2).  The complexity of the test might dictate whether this is viable or not, the fallback would be for the custom test to be developed in python, Java, or golang as discussed in this proposal.
```

Comment:
We should add an issue to track ansible and helm default tests. 

For ansible I assume something based on molecule, that can get the scorecard output?

/cc @fabianvf 

Not Blocking